### PR TITLE
tests/provider: Fix hardcoded ARN (CloudTrail)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #15662

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

The output from acceptance testing (GovCloud):

```
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (15.42s)
--- PASS: TestAccAWSCloudTrailServiceAccount_Region (15.42s)
--- PASS: TestAccAWSCloudTrail_serial (837.99s)
    --- PASS: TestAccAWSCloudTrail_serial/Trail (837.99s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/includeGlobalServiceEvents (42.29s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/eventSelector (155.27s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/insightSelector (42.69s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/basic (74.04s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/cloudwatch (85.99s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/enableLogging (108.30s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/isMultiRegion (104.89s)
        --- SKIP: TestAccAWSCloudTrail_serial/Trail/isOrganization (1.95s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/logValidation (73.81s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/kmsKey (41.81s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/tags (106.96s)
```

The output from acceptance testing (commercial):

```
--- PASS: TestAccAWSCloudTrailServiceAccount_Region (9.74s)
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (9.78s)
--- FAIL: TestAccAWSCloudTrail_serial (727.57s)
    --- FAIL: TestAccAWSCloudTrail_serial/Trail (727.57s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/includeGlobalServiceEvents (40.03s)
        --- SKIP: TestAccAWSCloudTrail_serial/Trail/isOrganization (0.86s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/logValidation (68.41s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/kmsKey (38.69s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/tags (100.21s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/eventSelector (153.04s)
        --- FAIL: TestAccAWSCloudTrail_serial/Trail/basic (53.20s)
        --- FAIL: TestAccAWSCloudTrail_serial/Trail/enableLogging (53.72s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/isMultiRegion (101.63s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/insightSelector (39.56s)
        --- PASS: TestAccAWSCloudTrail_serial/Trail/cloudwatch (78.21s)
```
**Failures are unrelated and due to permissions in the commercial alt account (I am "not authorized to perform: cloudtrail:StopLogging").

### CODE REVIEW

**Comment by bflad** [comment]
Extraneous go.mod/go.sum changes should be removed 👍

**Review by bflad — CHANGES_REQUESTED** [review]
Merge conflict 😉

**Comment by YakDriver** [comment]
What in the world has go done? Fixed.

File: aws/resource_aws_cloudtrail_test.go [hunk 1]
```diff
@@ -732,21 +734,23 @@ resource "aws_s3_bucket" "foo" {
 }
 POLICY
 }
-`, cloudTrailRandInt, cloudTrailRandInt, cloudTrailRandInt, cloudTrailRandInt)
+`, cloudTrailRandInt)
```
Comment (anGie44) [inline aws/resource_aws_cloudtrail_test.go:737]: 🙌 🙌

**Review by anGie44 — APPROVED** [review]
LGTM 🚀

File: aws/resource_aws_cloudtrail_test.go (hunk 1 above)
Comment (YakDriver) [reply aws/resource_aws_cloudtrail_test.go:737]: 💯

**Pull request merged**